Release notes for 2.18.0 (#12785)
PrefectHQ/prefect: RELEASE-NOTES.md

- Simplify quickstart — https://github.com/PrefectHQ/prefect/pull/12725
- Add `.serve`, `.deploy`, and composite trigger examples to deployment triggers docs — https://github.com/PrefectHQ/prefect/pull/12743
- Update automations images — https://github.com/PrefectHQ/prefect/pull/12752
- Simplify tutorial — https://github.com/PrefectHQ/prefect/pull/12765
- Remove disclaimer for Python 3.12 experimental support — https://github.com/PrefectHQ/prefect/pull/12771
- Clarify deployment trigger examples — https://github.com/PrefectHQ/prefect/pull/12782
- Remove Prefect-managed integration libraries to be archived from the integrations catalog — https://github.com/PrefectHQ/prefect/pull/12781
- Fix broken link to push work pool guide — https://github.com/PrefectHQ/prefect/pull/12748
- Fix minor restructure to improve legibility of work pools tutorial — https://github.com/PrefectHQ/prefect/pull/12747
- Fix `typing` import and typos in tasks tutorial — https://github.com/PrefectHQ/prefect/pull/12746
- Simplify installation — https://github.com/PrefectHQ/prefect/pull/12772
- Fix import syntax in `variables.Variable` example — https://github.com/PrefectHQ/prefect/pull/12727

> - Fix typo in How-to Guide document — https://github.com/PrefectHQ/prefect/pull/12761

Reviewer: ```suggestion
- Fix typo in How-to Guide — https://github.com/PrefectHQ/prefect/pull/12761
```

- Implementing the `pause-automation` and `resume-automation` actions — https://github.com/PrefectHQ/prefect/pull/12738
- Add automations CLI — https://github.com/PrefectHQ/prefect/pull/12754
- Rename `prefect-cloud.*` events and labels to `prefect.*` — https://github.com/PrefectHQ/prefect/pull/12755
- Add ability to emit events to an ephemeral Prefect server — https://github.com/PrefectHQ/prefect/pull/12762
- Disable `events` and `automations` API routes when experimental events setting is not enabled — https://github.com/PrefectHQ/prefect/pull/12777
- Add compatibility tests for client and server triggers and actions — https://github.com/PrefectHQ/prefect/pull/12778
- Disable the automations integration flows for Prefect Cloud — https://github.com/PrefectHQ/prefect/pull/12784
- Implement the work pool and work queue actions — https://github.com/PrefectHQ/prefect/pull/12735
- Add helper functions for creating an events client or subscriber — https://github.com/PrefectHQ/prefect/pull/12759
- Add default posture to `EventTrigger` schema — https://github.com/PrefectHQ/prefect/pull/12764

>  - Fix writing events for SQLite + SQLAlchemy<2 — https://github.com/PrefectHQ/prefect/pull/12679

Reviewer: ```suggestion
- Fix writing events for SQLite + SQLAlchemy<2 — https://github.com/PrefectHQ/prefect/pull/12679
```

### Documentation
- Update `prefect.yaml` example in work pools concepts page — https://github.com/PrefectHQ/prefect/pull/12695

> - Fix typo in Quickstart — https://github.com/PrefectHQ/prefect/pull/12729

Reviewer: ```suggestion
- Fix typo in quickstart — https://github.com/PrefectHQ/prefect/pull/12729
```

- Implement the `pause-deployment` and `resume-deployment` actions — https://github.com/PrefectHQ/prefect/pull/12733
- Add `/events/filter` and `/events/count-by` route trees — https://github.com/PrefectHQ/prefect/pull/12736
- Allow for creating automations via deployments when experimental events is on — https://github.com/PrefectHQ/prefect/pull/12701
- Add ability to stream out events via websocket — https://github.com/PrefectHQ/prefect/pull/12744
- Implementing the `pause-automation` and `resume-automation` actions — https://github.com/PrefectHQ/prefect/pull/12738
- Add an Automations CLI — https://github.com/PrefectHQ/prefect/pull/12754
- Rename `prefect-cloud.*` events and labels to `prefect.*` — https://github.com/PrefectHQ/prefect/pull/12755
- Add ability to emit events to an ephemeral Prefect server — https://github.com/PrefectHQ/prefect/pull/12762
- Disable `events` and `automations` API routes when experimental events setting is not enabled — https://github.com/PrefectHQ/prefect/pull/12777
- Add compatibility tests for client and server triggers and actions — https://github.com/PrefectHQ/prefect/pull/12778

> - Disable the automations integration flows for Prefect Cloud — https://github.com/PrefectHQ/prefect/pull/12784

Reviewer 2: In the past couple of releases we've been further dividing these PRs into events and pydantic for people who are interested in in progress features.

# Prefect Release Notes

## Release 2.18.0

### Breaking Changes
- Remove deprecated ability to run `prefect deploy` with `deployment.yaml` file — https://github.com/PrefectHQ/prefect/pull/12731
- Remove deprecated ability to pass `-f/--flow` as option to `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/12732
- Remove deprecated `projects` from `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/12737
- Remove deprecated `--ci` option from `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/12740

### Enhancements

> - Introduce self-validating types — https://github.com/PrefectHQ/prefect/pull/12707

Reviewer 3: this can go in Pydantic section

> - Allow returning to account selection in `prefect cloud login` and `workspace set` — https://github.com/PrefectHQ/prefect/pull/12717

Reviewer 3: ```suggestion
- Improve account selection in `prefect cloud login` and `workspace set` — https://github.com/PrefectHQ/prefect/pull/12717
```

> - Add default posture to `EventTrigger` schema — https://github.com/PrefectHQ/prefect/pull/12764

Reviewer 3: probably can go in events section

### Fixes

> - Better flow validation error — https://github.com/PrefectHQ/prefect/pull/12715

Reviewer 3: ```suggestion
- Raise clearer flow validation error — https://github.com/PrefectHQ/prefect/pull/12715
```

- Fix import syntax in `variables.Variable` example — https://github.com/PrefectHQ/prefect/pull/12727
- Fix typo in How-to Guide document — https://github.com/PrefectHQ/prefect/pull/12761

> - Fix writing events for SQLite + SQLAlchemy<2 — https://github.com/PrefectHQ/prefect/pull/12679

Reviewer 3: events section?

- Fix a flake with the SQLite version check test — https://github.com/PrefectHQ/prefect/pull/12690
- Fix a test flake in flow run lateness — https://github.com/PrefectHQ/prefect/pull/12698

> - Remove `version` from `docker-compose.yml`, as it is now deprecated — https://github.com/PrefectHQ/prefect/pull/12706

Reviewer 3: may be maintenance and therefore not needed?

> - Remove `version` from `docker-compose.yml`, as it is now deprecated — https://github.com/PrefectHQ/prefect/pull/12706

Reviewer 3: maintenance fixes probably don't need to be in the release

> - Fix types NonNegativeDuration / PositiveDuration — https://github.com/PrefectHQ/prefect/pull/12711

Reviewer 3: probably pydantic?

- Exclude job_variables when exclude=None — https://github.com/PrefectHQ/prefect/pull/12712

> - Remove all infra overrides-related flags — https://github.com/PrefectHQ/prefect/pull/12742

Reviewer 3: ```suggestion
- Remove experimental flags on infrastructure overrides — https://github.com/PrefectHQ/prefect/pull/12742
```

#### Events and Automations
- Implement the `run-deployment` automation action — https://github.com/PrefectHQ/prefect/pull/12677
- Implement the `send-notification` action — https://github.com/PrefectHQ/prefect/pull/12693
- Make `TriggeredAction.firing` required — https://github.com/PrefectHQ/prefect/pull/12697
- Add an Actions service — https://github.com/PrefectHQ/prefect/pull/12699
- Implement the `call-webhook` action and adds all Action client-side schemata — https://github.com/PrefectHQ/prefect/pull/12728
- Implement `change-flow-run-state`, `cancel-flow-run`, and `suspend-flow-run` — https://github.com/PrefectHQ/prefect/pull/12730
- Add functions for querying and counting events — https://github.com/PrefectHQ/prefect/pull/12696
- Implement the `pause-deployment` and `resume-deployment` actions — https://github.com/PrefectHQ/prefect/pull/12733
- Add `/events/filter` and `/events/count-by` route trees — https://github.com/PrefectHQ/prefect/pull/12736
- Allow for creating automations via deployments when experimental events is on — https://github.com/PrefectHQ/prefect/pull/12701
- Add ability to stream out events via websocket — https://github.com/PrefectHQ/prefect/pull/12744

> - Implementing the `pause-automation` and `resume-automation` actions — https://github.com/PrefectHQ/prefect/pull/12738

Reviewer 3: ```suggestion
- Implement the `pause-automation` and `resume-automation` actions — https://github.com/PrefectHQ/prefect/pull/12738
```

> - Add an Automations CLI — https://github.com/PrefectHQ/prefect/pull/12754

Reviewer 3: ```suggestion
- Add automations CLI — https://github.com/PrefectHQ/prefect/pull/12754
```

- Rename `prefect-cloud.*` events and labels to `prefect.*` — https://github.com/PrefectHQ/prefect/pull/12755
- Add ability to emit events to an ephemeral Prefect server — https://github.com/PrefectHQ/prefect/pull/12762
- Disable `events` and `automations` API routes when experimental events setting is not enabled — https://github.com/PrefectHQ/prefect/pull/12777
- Add compatibility tests for client and server triggers and actions — https://github.com/PrefectHQ/prefect/pull/12778
- Disable the automations integration flows for Prefect Cloud — https://github.com/PrefectHQ/prefect/pull/12784

> - Implement the work pool and work queue actions — https://github.com/PrefectHQ/prefect/pull/12735

Reviewer: ```suggestion
- Add pause and resume the work pool and work queue actions — https://github.com/PrefectHQ/prefect/pull/12735
```

- Add helper functions for creating an events client or subscriber — https://github.com/PrefectHQ/prefect/pull/12759

### Documentation

> - Update example to `prefect.yaml` in  work-pools concepts page — https://github.com/PrefectHQ/prefect/pull/12695

Reviewer 3: ```suggestion
- Update `prefect.yaml` example in work pools concepts page — https://github.com/PrefectHQ/prefect/pull/12695
```

- Fix stray link — https://github.com/PrefectHQ/prefect/pull/12721
- Add hyphen — https://github.com/PrefectHQ/prefect/pull/12724

> - Fix typo — https://github.com/PrefectHQ/prefect/pull/12722

Reviewer 3: these may not be worth including, esp if we don't feel like being descriptive

Reviewer: +1

> - Remove duplicative resilient — https://github.com/PrefectHQ/prefect/pull/12723

Reviewer 3: same with this one

- Fix typo in Quickstart — https://github.com/PrefectHQ/prefect/pull/12729

> - Simplify-api-connection — https://github.com/PrefectHQ/prefect/pull/12725

Reviewer 3: ```suggestion
- Simplify quickstart — https://github.com/PrefectHQ/prefect/pull/12725
```

- Add `.serve`, `.deploy`, and composite trigger examples to deployment triggers docs — https://github.com/PrefectHQ/prefect/pull/12743
- Update automations images — https://github.com/PrefectHQ/prefect/pull/12752
- Simplify tutorial — https://github.com/PrefectHQ/prefect/pull/12765
- Remove disclaimer for Python 3.12 experimental support — https://github.com/PrefectHQ/prefect/pull/12771
- Clarify deployment trigger examples — https://github.com/PrefectHQ/prefect/pull/12782
- Remove Prefect-managed integration libraries to be archived from the integrations catalog — https://github.com/PrefectHQ/prefect/pull/12781

> - Fix broken link to Push Work Pool Guide — https://github.com/PrefectHQ/prefect/pull/12748

Reviewer 3: ```suggestion
- Fix broken link to push work pool guide — https://github.com/PrefectHQ/prefect/pull/12748
```

> - Fix minor restructure to improve legibility of `work pools` tutorial — https://github.com/PrefectHQ/prefect/pull/12747

Reviewer 3: ```suggestion
- Fix minor restructure to improve legibility of work pools tutorial — https://github.com/PrefectHQ/prefect/pull/12747
```

> - Fix `typing` import and typos in Tasks tutorial — https://github.com/PrefectHQ/prefect/pull/12746

Reviewer 3: ```suggestion
- Fix `typing` import and typos in tasks tutorial — https://github.com/PrefectHQ/prefect/pull/12746
```

- Simplify installation — https://github.com/PrefectHQ/prefect/pull/12772

> 
> ### Uncategorized

Reviewer 3: ```suggestion
```

## New Contributors
* @hboehmer-IW made their first contribution in https://github.com/PrefectHQ/prefect/pull/12721
* @avriiil made their first contribution in https://github.com/PrefectHQ/prefect/pull/12748
* @takashimakazuki made their first contribution in https://github.com/PrefectHQ/prefect/pull/12761
  
### Integrations
- Add support for a capacity provider — https://github.com/PrefectHQ/prefect-aws/pull/407
- Improve error handling for task creation — https://github.com/PrefectHQ/prefect-aws/pull/406

> ### Contributors
> - @avriiil
> - @hboehmer-IW
> - @takashimakazuki

Reviewer 3: these are dupes since they're all new contributors so we can remove this
```suggestion
```

**All changes**: https://github.com/PrefectHQ/prefect/compare/2.17.1...2.18.0

> 

Reviewer 3: ```suggestion
```
